orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -682,6 +682,29 @@ at::Tensor& out) {
   return out;
 }
 
+const at::Tensor& resize_(
+    const at::Tensor& self,
+    at::IntArrayRef size,
+    c10::optional<at::MemoryFormat> optional_memory_format) {
+  // TODO: handle named tensor...
+  // TODO: handle optional_memory_format
+
+  ORT_LOG_FN(self, size, optional_memory_format);
+  assert_tensor_supported(self);
+
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto self_ort_value = create_ort_value(invoker, self);
+  auto& self_ort_tensor = self_ort_value.Get<onnxruntime::Tensor>();
+  auto* self_impl = dynamic_cast<ORTTensorImpl*>(self.unsafeGetTensorImpl());
+
+  onnxruntime::Tensor::InitOrtValue(self_ort_tensor.DataType(), onnxruntime::TensorShape(size.vec()),
```

Comment:
In this case, `resize_` and `reshape` are slightly different:

[resize_](https://pytorch.org/docs/stable/generated/torch.Tensor.resize_.html?highlight=resize) can change the underlying storage:


> Resizes self tensor to the specified size. If the number of elements is larger than the current storage size, then the underlying storage is resized to fit the new number of elements. If the number of elements is smaller, the underlying storage is not changed. Existing elements are preserved but any new memory is uninitialized.

[reshape](https://pytorch.org/docs/stable/generated/torch.reshape.html#torch.reshape): returns tensor with same data and number of elements

> Returns a tensor with the same data and number of elements as input, but with the specified shape. When possible, the returned tensor will be a view of input. Otherwise, it will be a copy. Contiguous inputs and inputs with compatible strides can be reshaped without copying, but you should not depend on the copying vs. viewing behavior.